cmd/cluster/dynatrace/dtquery.go

```diff
@@ -0,0 +1,140 @@
+package dynatrace
+
+import (
+	"fmt"
+	"strings"
+)
+
+type DTQuery struct {
+	fragments  []string
+	finalQuery string
+}
+
+func (q *DTQuery) Init(hours int) *DTQuery {
+	q.fragments = []string{}
+
+	q.fragments = append(q.fragments, fmt.Sprintf("fetch logs, from:now()-%dh \n| filter matchesValue(event.type, \"LOG\")", hours))
+
+	return q
+}
+
+func (q *DTQuery) Cluster(mgmtClusterName string) *DTQuery {
+	q.fragments = append(q.fragments, fmt.Sprintf(" and matchesPhrase(dt.kubernetes.cluster.name, \"%s\")", mgmtClusterName))
+
+	return q
+}
+
+func (q *DTQuery) Namespaces(namespaceList []string, clusterID string, hcp bool) *DTQuery {
+	var nsQuery string
+	finalQuery := ""
+
+	if !hcp {
+		nsQuery = " and ("
+	}
+
+	if hcp && len(namespaceList) > 0 {
+		finalQuery += fmt.Sprintf(" and ( matchesPhrase(k8s.namespace.name, \"ocm-production-%s\")", clusterID)
+		nsQuery = " or "
+	}
+
+	for i, ns := range namespaceList {
+		nsQuery += fmt.Sprintf("matchesValue(k8s.namespace.name, \"%s\")", ns)
+		if i < len(namespaceList)-1 {
+			nsQuery += " or "
+		}
+	}
+	nsQuery += ")"
+	finalQuery += nsQuery
+
+	q.fragments = append(q.fragments, finalQuery)
+
+	return q
+}
+
+func (q *DTQuery) Nodes(nodeList []string) *DTQuery {
+	var nodeQuery string
+
+	nodeQuery = " and ("
+	for i, node := range nodeList {
+		nodeQuery += fmt.Sprintf("matchesValue(k8s.node.name, \"%s\")", node)
+		if i < len(nodeList)-1 {
+			nodeQuery += " or "
+		}
+	}
+	nodeQuery += ")"
+	q.fragments = append(q.fragments, nodeQuery)
+
+	return q
+}
+
+func (q *DTQuery) Pods(podList []string) *DTQuery {
+	var podQuery string
+
+	podQuery = " and ("
+	for i, pod := range podList {
+		podQuery += fmt.Sprintf("matchesValue(k8s.pod.name, \"%s\")", pod)
+		if i < len(podList)-1 {
+			podQuery += " or "
+		}
+	}
+	podQuery += ")"
+	q.fragments = append(q.fragments, podQuery)
+
+	return q
+}
+
+func (q *DTQuery) Containers(containerList []string) *DTQuery {
+	var containerQuery string
+
+	containerQuery = " and ("
+	for i, container := range containerList {
+		containerQuery += fmt.Sprintf("matchesValue(k8s.container.name, \"%s\")", container)
+		if i < len(containerList)-1 {
+			containerQuery += " or "
+		}
+	}
+	containerQuery += ")"
+	q.fragments = append(q.fragments, containerQuery)
+
+	return q
+}
+
+func (q *DTQuery) Status(statusList []string) *DTQuery {
+	var statusQuery string
+
+	statusQuery = " and ("
+	for i, status := range statusList {
+		statusQuery += fmt.Sprintf("matchesValue(status, \"%s\")", status)
+		if i < len(statusList)-1 {
+			statusQuery += " or "
+		}
+	}
+	statusQuery += ")"
+	q.fragments = append(q.fragments, statusQuery)
+
+	return q
+}
+
+func (q *DTQuery) ContainsPhrase(phrase string) *DTQuery {
+	q.fragments = append(q.fragments, " and contains(content,\""+phrase+"\")")
+
+	return q
+}
+
+func (q *DTQuery) Sort(order string) *DTQuery {
+	q.fragments = append(q.fragments, fmt.Sprintf("\n| sort timestamp %s", order))
+
+	return q
+}
```

Comment:
**Suggestion:** check for allowed strings. What if the user enters something that isn't "asc" or "desc"?